extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/TlsCertificateReloadUtils.java

```diff
@@ -0,0 +1,133 @@
+
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.KeyStoreOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.SSLOptions;
+
+/**
+ * Utility class to handle TLS certificate reloading.
+ */
+public class TlsCertificateReloadUtils {
+
+    public static long handleCertificateReloading(Vertx vertx, HttpServer server,
+            HttpServerOptions options, HttpConfiguration configuration) {
+        // Validation
+        if (configuration.ssl.certificate.reloadPeriod.isEmpty()) {
+            return -1;
+        }
+        if (configuration.ssl.certificate.reloadPeriod.get().toMillis() < 1000) {
+            throw new IllegalArgumentException(
+                    "Unable to configure TLS reloading - The reload period cannot be less than 1 second");
+        }
+        if (options == null) {
+            throw new IllegalArgumentException("Unable to configure TLS reloading - The HTTP server options were not provided");
+        }
+        SSLOptions ssl = options.getSslOptions();
+        if (ssl == null) {
+            throw new IllegalArgumentException("Unable to configure TLS reloading - TLS/SSL is not enabled on the server");
+        }
+
+        Logger log = Logger.getLogger(TlsCertificateReloadUtils.class);
+        return vertx.setPeriodic(configuration.ssl.certificate.reloadPeriod.get().toMillis(), id -> {
+
+            vertx.executeBlocking(new Callable<SSLOptions>() {
+                @Override
+                public SSLOptions call() throws Exception {
+                    // We are reading files - must be done on a worker thread.
+                    var c = reloadFileContent(ssl, configuration);
+                    if (c.equals(ssl)) { // No change, skip the update
+                        return null;
+                    }
+                    return c;
+                }
+            }, true)
+                    .flatMap(new Function<SSLOptions, Future<Boolean>>() {
+                        @Override
+                        public Future<Boolean> apply(SSLOptions res) {
+                            if (res != null) {
+                                return server.updateSSLOptions(res);
+                            } else {
+                                return Future.succeededFuture(false);
+                            }
+                        }
+                    })
+                    .onComplete(new Handler<AsyncResult<Boolean>>() {
+                        @Override
+                        public void handle(AsyncResult<Boolean> ar) {
+                            if (ar.failed()) {
+                                log.error("Unable to reload the TLS certificate, keeping the current one.", ar.cause());
```

Comment:
I agree this is a good default, but maybe what we do should be configurable by the user?

Reply:
Okay